activesupport/lib/active_support/values/time_zone.rb

```diff
@@ -513,14 +513,14 @@ def yesterday
     def utc_to_local(time)
       tzinfo.utc_to_local(time).yield_self do |t|
         ActiveSupport.utc_to_local_returns_utc_offset_times ?
-          t : Time.utc(t.year, t.month, t.day, t.hour, t.min, t.sec, t.sec_fraction)
+          t : Time.utc(t.year, t.month, t.day, t.hour, t.min, t.sec, t.sec_fraction * 1_000_000)
```

Comment:
FWIW, this is only relevant if someone calls this method explicitly. When Ruby calls this method (when TimeZone has been used as a `tz` or `in` argument) it will actually ignore the `sec_fraction` (the Time::tm object it passes to the method doesn't even implement it).

Reply:
Yes, you're right - that's what led me to track down https://github.com/ruby/ruby/pull/3821 but it causes a test to fail after the change so needs fixing.